portable/ThirdParty/GCC/Posix/port.c

```diff
@@ -163,13 +167,15 @@ StackType_t * pxPortInitialiseStack( StackType_t * pxTopOfStack,
     thread->pvParams = pvParameters;
     thread->xDying = pdFALSE;
 
+    /* Ensure ulStackSize is at least PTHREAD_STACK_MIN */
+    ulStackSize = (ulStackSize < PTHREAD_STACK_MIN) ? PTHREAD_STACK_MIN : ulStackSize;
+
     pthread_attr_init( &xThreadAttributes );
-    iRet = pthread_attr_setstack( &xThreadAttributes, pxEndOfStack, ulStackSize );
+    iRet = pthread_attr_setstacksize( &xThreadAttributes, ulStackSize );
```

Comment:
Correct.

The original intent was a good one, use the users stack for the POSIX port implementation. Unfortunately the minimum size means almost all user stacks will be too small, at least for the 32bit arm chips I've used FreeRTOS on its common to have 4k or 8k stacks but 16k stacks are less usual.

So the commit intends to resolve the stack reuse issue that valgrind catches, plus the unnecessary warnings issue, while still allowing the user to use very large stacks(>16k). I don't imagine this is a common use case but it can happen and only the main thread grows its stack on linux so its sometimes necessary.

I hope the commit log was also clear about the rational. I get that it's a change in functionality but imo there was little value in actually reusing the stack the user provided for this port.

Reply:
@Skptak thoughts? Does this help resolve the concern about the change in usage of the stack away from the user provided one?